SetupNode/ericomshield-setup-node.sh

```diff
@@ -137,10 +159,45 @@ make-leader-ip() {
 }
 
 
-make_machines_ready() {
+append-sshpass() {
+    if ! command_exists sshpass; then
+        sudo apt-get install -y --assume-yes sshpass
+    fi
+}
 
-    for ip in $MACHINE_IPS; do
 
+collect_machine_pass() {
+    read -s -p "Machine user Password: " MACHINE_USER_PASS
+}
+
+
+make_leader_hosts_record() {
+     IFS=':' read -r -a array <<< "$LEADER_IP"
+     CLEAN_IP="${array[0]}"
+     CLEAN_HOST=$(hostname)
+
+     echo "$CLEAN_IP    $CLEAN_HOST"
+}
+
+
+make_machines_ready() {
+    ssh-keygen -b 2048 -t rsa -f $CERTIFICATE_FILE -q -N ""
+    cert_name=$(basename $CERTIFICATE_FILE)
+    for ip in $MACHINE_IPS; do
+        echo '#################################################################### Prepare machine interactive #########################################'
+        sshpass -p$MACHINE_USER_PASS scp -o StrictHostKeyChecking=no "$CERTIFICATE_FILE.pub" $MACHINE_USER@$ip:~/authorized_keys
```

Comment:
I would quote all these variables like this: sshpass -p"$MACHINE_USER_PASS" scp -o StrictHostKeyChecking=no "$CERTIFICATE_FILE.pub" "$MACHINE_USER@$ip":~/authorized_keys